### ISSUE TITLE
AWS Go SDK can log to stdout, breaking some Terraform commands

### ISSUE THREAD

**Issue by geekofalltrades** [hashicorp/terraform-provider-aws#23620]
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Terraform CLI and Terraform AWS Provider Version

```
Terraform v1.1.4
on linux_amd64
+ provider registry.terraform.io/hashicorp/aws v3.74.1
```

### Affected Resource(s)

Not resource-specific.

### Terraform Configuration Files

Any.

### Expected Behavior

stdout is not polluted with SDK logs:

```
$ AWS_CSM_ENABLED=true AWS_CSM_PORT=31000 AWS_CSM_HOST=127.0.0.1 terraform output -json 2> /dev/null                                                                
{ # valid JSON output is produced
```

### Actual Behavior

stdout is polluted with SDK logs:

```
$ AWS_CSM_ENABLED=true AWS_CSM_PORT=31000 AWS_CSM_HOST=127.0.0.1 terraform output -json 2> /dev/null
2022/03/10 13:37:32 Enabling CSM                                                                   
{ # this is no longer valid JSON because the log message above is part of stdout
```

### Steps to Reproduce

1. Run Terraform with an AWS provider configured with environment variable `AWS_CSM_ENABLED=true`.

Ideally the AWS Go SDK would allow end-users to disable this logging, but that option doesn't seem to be available: https://github.com/aws/aws-sdk-go/issues/1611

Can the Terraform provider instead be careful to blackhole all of this logging or ensure that it goes to stderr?

**Comment by ewbankkit** [comment]
Relevant: https://docs.aws.amazon.com/sdk-for-go/api/aws/csm/.

**Comment by kristofferahl** [comment]
@geekofalltrades Did you find a workaround? I just hit this issue myself but don't want to disable CSM.

**Comment by geekofalltrades** [comment]
No. I ended up abandoning CSM and finding another way to complete my task.

**Comment by c4milo** [comment]
I just ran into this as well, very surprising and time consuming.

**Comment by c4milo** [comment]
I think it is worth mentioning that setting `TF_LOG=OFF` does not help.
